crypto/bindings/conversion.ts

```diff
@@ -98,3 +158,120 @@ function fieldsFromRustFlat(fieldBytes: Uint8Array): MlArray<Field> {
   }
   return [0, ...fields];
 }
+
+// affine
+
+type WasmAffine = wasmNamespace.WasmGVesta | wasmNamespace.WasmGPallas;
+type MakeAffine<A extends WasmAffine> = () => A & { ptr: number };
+
+function affineFromRust(pt: WasmAffine): OrInfinity {
+  if (pt.infinity) {
+    pt.free();
+    return 0;
+  } else {
+    let x = fieldFromRust(pt.x);
+    let y = fieldFromRust(pt.y);
+    pt.free();
+    return [0, [0, x, y]];
+  }
+}
+
+function affineToRust<A extends WasmAffine>(
+  pt: OrInfinity,
+  klass: MakeAffine<A>
+) {
+  var res = klass();
+  if (pt === 0) {
+    res.infinity = true;
+  } else {
+    res.x = fieldToRust(pt[1][1]);
+    res.y = fieldToRust(pt[1][2]);
+  }
+  return res;
+}
+
+// polycomm
+
+type WasmPolyComm = wasmNamespace.WasmFpPolyComm | wasmNamespace.WasmFqPolyComm;
+type WasmPolyCommClass = wasm['WasmFpPolyComm'] | wasm['WasmFqPolyComm'];
+
+function polyCommFromRust(
+  polyComm: WasmPolyComm,
+  klass: WrapperClass<WasmAffine>,
+  shouldFree: boolean
+): PolyComm {
+  let rustShifted = polyComm.shifted;
+  let rustUnshifted = polyComm.unshifted;
+  let mlShifted: MlOption<OrInfinity> =
+    rustShifted === undefined ? 0 : [0, affineFromRust(rustShifted)];
+  let mlUnshifted = mlArrayFromRustVector(
+    rustUnshifted,
+    klass,
+    affineFromRust,
+    shouldFree
+  );
+  return [0, mlUnshifted, mlShifted];
+}
+
+function polyCommToRust(
+  [, camlUnshifted, camlShifted]: PolyComm,
+  PolyComm: WasmPolyCommClass,
+  makeAffine: MakeAffine<WasmAffine>
+): WasmPolyComm {
+  let rustShifted =
+    camlShifted === 0 ? undefined : affineToRust(camlShifted[1], makeAffine);
```

Comment:
As discussed [here](https://github.com/o1-labs/o1js-bindings/pull/140#discussion_r1329991174), would require a change, even though OK as the first element of the ADT is 0.

Reply:
no -- camlShifted is actually supposed to be an `Option`. I don't want to use a helper here because that would require me to create an inline conversion function in this case, which is inefficient, and `polyCommToRust` is used a lot. So will stick to the existing code